pennylane/compiler/compiler.py

```diff
@@ -0,0 +1,136 @@
+
+    # Refresh the list of compilers
+    AvailableCompilers.names_entrypoints = {}
+
+    # Iterator packages entry-points with the 'pennylane.compilers' group name
+    for entry in pkg_resources.iter_entry_points("pennylane.compilers"):
+        module_name = entry.module_name
+        # Only need name of the parent module
+        module_name = module_name.split(".")[0]
+
+        if module_name not in AvailableCompilers.names_entrypoints:
+            AvailableCompilers.names_entrypoints[module_name] = {}
+        AvailableCompilers.names_entrypoints[module_name][entry.name] = entry
+
+
+def available_compilers() -> List[str]:
+    """Return the name of available compilers by refreshing the compilers
+    names and entry points.
+
+    **Example**
+
+    This method returns the name of installed compiler packages supported in
+    PennyLane. For example, after installing the ``pennylane-catalyst`` pacakge,
+
+    .. code-block:: python
+
+    >>> qml.compiler.available_compilers()
+    ['catalyst']
+    """
+
+    # This class updates the class variable names_entrypoints
+    _refresh_compilers()
+
+    return list(AvailableCompilers.names_entrypoints.keys())
+
+
+def available(name="catalyst") -> bool:
+    """Check the availability of the given compiler package.
+
+    It only refreshes the compilers names and entry points if the name
+    is not already stored. This reduces the number of re-importing
+    ``pkg_resources`` as it can be a very slow operation on systems
+    with a large number of installed packages.
+
+    Args:
+        name (str): name of the compiler package (Default is ``catalyst``)
+
+    Return:
+        bool: ``True`` if the compiler package is installed on the system
+
+    **Example**
+
+    Before installing the ``pennylane-catalyst`` package:
+
+    .. code-block:: python
+
+    >>> qml.compiler.available("catalyst")
+    False
+
+    After installing the ``pennylane-catalyst`` package:
+
+    >>> qml.compiler.available("catalyst")
+    True
+    """
+
+    if name not in AvailableCompilers.names_entrypoints:
+        # This class updates the class variable names_entrypoints
+        _refresh_compilers()
+
+    return name in AvailableCompilers.names_entrypoints
+
+
+def active(name="catalyst") -> bool:
+    """Check whether the caller is inside a QJIT evaluation context.
+
+    Args:
+        name (str): name of the compiler package (Default is ``catalyst``)
+
+    Return:
+        bool: True if the caller is inside a QJIT evaluation context
+
+    **Example**
+
+    In the JIT compilation of PennyLane programs, this helper method checks
+    the status of the compilation. For example, Catalyst captures Python
+    programs by tracing and we can tell if the caller is inside the Catalyst
+    tracer context manager or not but calling this method.
+
+    This method is practically useful in implementing quantum operations to
+    correctly call the interpreter or compiler equivalent functions.
```

Comment:
@maliasadi recommend replacing the text explanation with a code example if possible :)

Reply:
I couldn't come up with a simple code example to show case this method. However, there are some use cases in PR #4698. This method can be used within functions that are evaluated by the tracer so the example for this will require importing a few tracing methods from JAX.

Reply:
I was thinking too hard for this example :laughing: Added a simple example :))